### PR TITLE
[SPARK-36049][SQL] Remove IntervalUnit

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Remove IntervalUnit

### Why are the changes needed?
Clean code

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Not need

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
FYI @MaxGekk

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala [hunk 1]
```diff
@@ -774,19 +747,32 @@ object IntervalUtils {
...
+  def intervalUnits: Seq[UTF8String] = {
+    Seq(yearStr,
+      monthStr,
+      weekStr,
+      dayStr,
+      hourStr,
+      minuteStr,
+      secondStr,
+      millisStr,
+      microsStr)
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala:774]: Is it used somewhere except of tests? Just wonder why did you move it from `ExpressionParserSuite`?

**Review by MaxGekk — COMMENTED** [review]
LGTM in general except of a comment.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala (hunk 1 above)
Comment (AngersZhuuuu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala:774]: > Is it used somewhere except of tests? Just wonder why did you move it from `ExpressionParserSuite`?

Move back, how about current?

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
@AngersZhuuuu Could you re-trigger GitHub actions, please.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master/3.2.
Thank you, @AngersZhuuuu .
